subsys/modbus/Kconfig

```diff
@@ -79,6 +79,62 @@ config MODBUS_FC08_DIAGNOSTIC
 	help
 	  Enable function code 08 Diagnostic support
 
+config MODBUS_FC01_COIL_RD
+	bool "FC01 single coil read support"
+	depends on MODBUS_SERVER
+	default y
+	help
+	  Enable function code 01 single coil read support
+
+config MODBUS_FC02_DI_RD
+	bool "FC02 single discrete input read support"
+	depends on MODBUS_SERVER
+	default y
+	help
+	  Enable function code 02 single discrete input read support
+
+config MODBUS_FC03_HOLDING_REG_RD
+	bool "FC03 multiple holding register read support"
+	depends on MODBUS_SERVER
+	default y
+	help
+	  Enable function code 03 multiple holding register read support
+
+config MODBUS_FC04_IN_REG_RD
+	bool "FC04 multiple input register read support"
+	depends on MODBUS_SERVER
+	default y
+	help
+	  Enable function code 04 multiple input register read support
+
+config MODBUS_FC05_COIL_WR
+	bool "FC05 single coil write support"
+	depends on MODBUS_SERVER
+	default y
+	help
+	  Enable function code 05 single coil write support
+
+config MODBUS_FC06_HOLDING_REG_WR
+	bool "FC06 single holding register write support"
+	depends on MODBUS_SERVER
+	default y
+	help
+	  Enable function code 06 single holding register write support
+
+config MODBUS_FC15_COILS_WR
+	bool "FC15 multiple coils write support"
+	depends on MODBUS_SERVER
+	default y
+	help
+	  Enable function code 15 multiple coils write support
+
+config MODBUS_FC16_HOLDING_REGS_WR
+	bool "FC16 multiple holding register write support"
+	depends on MODBUS_SERVER
+	default y
+	help
+	  Enable function code 16 multiple holding register write support
+
```

Comment:
Obviously you do not like a flexible implementation that uses only the code and therefore the footprint that is required.
You also allow any other FC to be implemented in a custom manner, but here you insist on using your implementation.

We have it downstream with the proposed flexibility anyway and thought to give it back to the project.

It is out there now.
I cannot put more effort in so far fruitless discussions (#73237).

Maybe some other voices can convince.

Otherwise I withdraw and we kept it downstream.